sqrl-planner/sqrl-planner-local/src/main/java/com/datasqrl/plan/global/DAGBuilder.java

```diff
@@ -51,24 +51,30 @@ public SqrlDAG build(Collection<AnalyzedAPIQuery> queries,
     Multimap<SqrlNode, SqrlNode> dagInputs = HashMultimap.create();
     //1. Add all queries as sinks
     List<ExecutionStage> readStages = pipeline.getReadStages();
-    errors.checkFatal(!readStages.isEmpty(), "Configured Pipeline does not include"
-        + " any read stages to execute API queries: %s",pipeline);
+    if (readStages.isEmpty()) {
```

Comment:
Can you make this a !readStages.isEmpty() check around the for loop so we don't have to repeat that code?